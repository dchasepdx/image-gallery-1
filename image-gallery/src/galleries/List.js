import React from 'react';
import PropTypes from 'prop-types';


    export default function List (props) {
  
        return (
            <div>
                {props.images.map(image => (
                    <ul className="list-view" key={image.url}>
                        <li>{image.title} </li>
                        <li>{image.description} </li>
                        <li>{image.url} </li>
                    </ul>))}
            </div>


        )
    }

    List.proptypes = {
images:PropTypes.array.isRequired
};