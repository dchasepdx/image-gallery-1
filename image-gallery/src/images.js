import images from './imagesData';

export default {
    get() { return Promise.resolve(images) }
}