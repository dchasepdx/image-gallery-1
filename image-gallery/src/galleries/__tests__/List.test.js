import React from 'react';
import ReactDOM from 'react-dom';
import List from '../List';
import renderer from 'react-test-renderer';
import images from '../../imagesData';



it('renders List view without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<List  images={images}/>, div);
});

it('renders a snapshot', () => {
  const tree = renderer.create(<List images={images}/>).toJSON();
  expect(tree).toMatchSnapshot();
});