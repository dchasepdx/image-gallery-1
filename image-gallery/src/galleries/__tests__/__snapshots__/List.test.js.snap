// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a snapshot 1`] = `
<div>
  <ul
    className="list-view"
  >
    <li>
      Cute Bunny
       
    </li>
    <li>
      Isn't it fuzzy-wuzzy cutest thing you've ever seen?
       
    </li>
    <li>
      http://f.cl.ly/items/3g3J1G0w122M360w380O/3726490195_f7cc75d377_o.jpg
       
    </li>
  </ul>
  <ul
    className="list-view"
  >
    <li>
      Cuter Bunny
       
    </li>
    <li>
      Butterscotty and sweet!!!
       
    </li>
    <li>
      http://static.boredpanda.com/blog/wp-content/uploads/2015/09/cute-bunnies-25__605.jpg
       
    </li>
  </ul>
  <ul
    className="list-view"
  >
    <li>
      Cutest Bunny
       
    </li>
    <li>
      snow bunny stare!
       
    </li>
    <li>
      http://static.boredpanda.com/blog/wp-content/uploads/2015/09/cute-bunnies-110__605.jpg
       
    </li>
  </ul>
</div>
`;